lib/models/project.js

```diff
@@ -75,14 +76,38 @@ class Project {
     instrumentation.project = this;
 
     this.emberCLIVersion = emberCLIVersion;
+
     this._nodeModulesPath = null;
+
+    if (this.cli && this.cli._packageInfoCache) {
+      this._packageInfoCache = this.cli._packageInfoCache;
+    } else {
+      this._packageInfoCache = new PackageInfoCache(this.ui);
+    }
+
+    // we're not dealing with NULL_PROJECT (note that it has not
+    // be set yet, so we can't compare to that var.)
+    this._packageInfo = this._packageInfoCache.loadProject(this);
+
+    // force us to use the real path as the root.
+    if (this.root !== this._packageInfo.realPath) {
```

Comment:
Yes, on the 'would it make more sense' part. At one point I was finding some weird instability, so I added an if and a log message. I'd rather that we assign root from the packageInfo instance, since that's already gone through the fs.realPathSync() tests itself, for speed. Are you okay with just .     this.root = this._packageInfo.realPath .   instead?  That's what I had before.